test/integration/consul-container/test/basic/connect_service_test.go

```diff
@@ -31,6 +31,7 @@ func TestBasicConnectService(t *testing.T) {
 	libassert.AssertUpstreamEndpointStatus(t, adminPort, "static-server.default", "HEALTHY", 1)
 	libassert.GetEnvoyListenerTCPFilters(t, adminPort)
 
+	libassert.AssertContainerState(t, clientService, "running")
```

Comment:
Do you think moving this assert to inside `HTTPServiceEchoes` would avoid duplicated assertion in all places calling `HTTPServiceEchoes`? To achieve this, we may pass the clientservice as an additional parameter to `HTTPServiceEchoes`, but I think this makes the `HTTPServiceEchoes` more readable.

Reply:
We'll need to add 2 parameters to `HTTPServiceEchoes`. `ClientService` and expected container status. I have the functions split so that HTTPServiceEchoes is strictly making HTTP calls and not performing multiple actions